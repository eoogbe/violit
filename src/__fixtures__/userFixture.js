/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const username = 'fred-astaire';

export const password = 'password123';

export const credentials = {
  username,
};

export const authUrl = 'http://localhost:4000/api/auth';

export const usersUrl = 'http://localhost:4000/api/users';

export const userUrl = 'http://localhost:4000/api/users/fred-astaire';

export const authApi = [
  {
    href: authUrl,
    data: [
      { name: 'username', value: username },
    ],
  }
];

export const userApi = [
  {
    href: userUrl,
    data: [
      { name: 'username', value: username },
    ],
  }
];

export const credentialsFetch = {
  url: authUrl,
  method: 'GET',
  status: 200,
  body: {
    items: userApi,
  },
};

export const loginFetch = {
  url: authUrl,
  method: 'POST',
  status: 201,
  body: {
    items: userApi,
  },
};

export const logoutFetch = {
  url: authUrl,
  method: 'DELETE',
  status: 200,
};

export const signupFetch = {
  url: usersUrl,
  method: 'POST',
  status: 201,
  body: {
    items: userApi,
  },
};

export const hasUserFetch = {
  url: userUrl,
  method: 'HEAD',
  status: 200,
};

export const doesNotHaveUserFetch = {
  url: userUrl,
  method: 'HEAD',
  status: 404,
};
