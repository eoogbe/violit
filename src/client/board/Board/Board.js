/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { MdAddBox } from 'react-icons/md';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { ThreadList } from 'client/threads';
import { theme, useTitle } from 'client/core';
import { LoadingContainer } from 'client/loading';
import * as api from '../api';

const StyledBoard = styled.article`
  font-size: 1rem;
  padding: ${({ theme }) => theme.spacingMd};
  position: relative;
`;

const Header = styled.header`
  font-size: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  border-bottom: 1px solid ${({ theme }) => theme.gray200};
`;

const Name = styled.h2`
  font-size: 1rem;
  margin: ${({ theme }) => theme.spacingSm} 0;
`;

/** The screen for a subforum. */
const Board = () => {
  useTitle();

  return (
    <StyledBoard>
      <Header>
        <LoadingContainer
          loadData={api.loadBoard}
          action="loading the board"
          render={({ name }) => (
            <Name>{name}</Name>
          )}
        />
        <Link to="/new-thread">
          {/* SVG title added for a11y. */}
          <MdAddBox
            color={theme.primaryColor}
            size="1.5em"
            attr={{ title: 'New thread', viewBox: '0 0 24 24' }}
          />
        </Link>
      </Header>
      <ThreadList />
    </StyledBoard>
  );
};

export default Board;
