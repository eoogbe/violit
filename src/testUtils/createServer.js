/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import cookieParser from 'cookie-parser';
import express from 'express';
import routes from 'server/api/routes';

/** Creates a test server for the specified `router`. */
export default (router, locals) => {
  const app = express();
  app.use(cookieParser());
  app.use(express.json());
  app.use((req, res, next) => {
    Object.assign(res.locals, { routes }, locals);
    next();
  });
  app.use(router());
  app.use((err, req, res, next) => {
    res.status(500).json({ error: err.message });
  });
  return app;
};
