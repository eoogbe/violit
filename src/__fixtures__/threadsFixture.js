/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as USER from './userFixture';

export const thread1 = {
  id: 'thread1',
  slug: 'lorem-ipsum',
  dateCreated: '2019-09-29T01:23:45.679Z',
  dateDeleted: '2019-09-29T02:00:45.679Z',
  headline: 'Lorem ipsum',
  articleBody: 'Dolor sit amet, consectetur adipiscing elit.',
  author: USER.username,
  title: 'Lorem ipsum | Violit',
  path: '/threads/lorem-ipsum',
  url: 'http://localhost:4000/api/threads/lorem-ipsum',
  deleteUrl: 'http://localhost:4000/api/threads/lorem-ipsum/delete',
  commentsUrl: 'http://localhost:4000/api/threads/lorem-ipsum/comments',
};

export const thread2 = {
  id: 'thread2',
  slug: 'aliquip-ex-ea',
  dateCreated: '2019-09-28T09:08:07.654Z',
  dateDeleted: '2019-09-28T09:08:08.654Z',
  headline: 'Aliquip ex ea',
  articleBody: null,
  author: USER.username,
  title: 'Aliquip ex ea | Violit',
  path: '/threads/aliquip-ex-ea',
  url: 'http://localhost:4000/api/threads/aliquip-ex-ea',
  deleteUrl: 'http://localhost:4000/api/threads/aliquip-ex-ea/delete',
  commentsUrl: 'http://localhost:4000/api/threads/aliquip-ex-ea/comments',
};

export const thread3 = {
  id: 'thread3',
  slug: 'commodo-consequat',
  dateCreated: '2019-09-27T03:21:09.876Z',
  dateDeleted: '2019-09-27T04:21:09.876Z',
  headline: 'Commodo consequat',
  articleBody: null,
  author: USER.username,
  title: 'Commodo consequat | Violit',
  path: '/threads/commodo-consequat',
  url: 'http://localhost:4000/api/threads/commodo-consequat',
  deleteUrl: 'http://localhost:4000/api/threads/commodo-consequat/delete',
  commentsUrl: 'http://localhost:4000/api/threads/commodo-consequat/comments',
};

export const createInput = {
  headline: thread1.headline,
  articleBody: thread1.articleBody,
};

export const normalized = {
  thread1: {
    id: 'thread1',
    slug: thread1.slug,
    dateCreated: thread1.dateCreated,
    dateDeleted: null,
    headline: thread1.headline,
    articleBody: thread1.articleBody,
    author: thread1.author,
  },
  thread2: {
    id: 'thread2',
    slug: thread2.slug,
    dateCreated: thread2.dateCreated,
    dateDeleted: thread2.dateDeleted,
    headline: thread2.headline,
    articleBody: thread2.articleBody,
    author: thread2.author,
  },
  thread3: {
    id: 'thread3',
    slug: thread3.slug,
    dateCreated: thread3.dateCreated,
    dateDeleted: null,
    headline: thread3.headline,
    articleBody: thread3.articleBody,
    author: thread3.author,
  },
};

export const unnormalized = [
  {
    id: 'thread1',
    slug: thread1.slug,
    dateCreated: thread1.dateCreated,
    headline: thread1.headline,
    articleBody: thread1.articleBody,
    author: thread1.author,
  },
  {
    id: 'thread3',
    slug: thread3.slug,
    dateCreated: thread3.dateCreated,
    headline: thread3.headline,
    articleBody: thread3.articleBody,
    author: thread3.author,
  },
];

export const api = [
  {
    href: thread1.url,
    data: [
      { name: 'id', value: 'thread1' },
      { name: 'slug', value: thread1.slug },
      { name: 'dateCreated', value: thread1.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'headline', value: thread1.headline },
      { name: 'articleBody', value: thread1.articleBody },
      { name: 'author', value: thread1.author },
    ],
    links: [
      { rel: 'deleteAction', href: thread1.deleteUrl },
      { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
      {
        rel: 'comment',
        href: thread1.commentsUrl,
      },
    ],
  },
  {
    href: thread3.url,
    data: [
      { name: 'id', value: 'thread3' },
      { name: 'slug', value: thread3.slug },
      { name: 'dateCreated', value: thread3.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'headline', value: thread3.headline },
      { name: 'articleBody', value: '' },
      { name: 'author', value: thread3.author },
    ],
    links: [
      { rel: 'deleteAction', href: thread3.deleteUrl },
      { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
      {
        rel: 'comment',
        href: thread3.commentsUrl,
      },
    ],
  },
];

export const deleteThreadApi = [
  {
    href: thread1.deleteUrl,
    data: [
      { name: 'actionStatus', value: 'completedActionStatus' },
      { name: 'endTime', value: thread1.dateDeleted },
    ],
    links: [
      { rel: 'object', href: thread1.url },
      { rel: 'targetCollection', href: 'http://localhost:4000/api/board' },
    ],
  },
];

export const threadsFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'GET',
  status: 200,
  body: {
    items: api,
  },
};

export const threadFetch1 = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [api[0]],
  },
};

export const threadWithoutArticleBodyFetch = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [
      {
        href: thread1.url,
        data: [
          { name: 'id', value: 'thread1' },
          { name: 'slug', value: thread1.slug },
          { name: 'dateCreated', value: thread1.dateCreated },
          { name: 'dateDeleted', value: '' },
          { name: 'headline', value: thread1.headline },
          { name: 'articleBody', value: '' },
          { name: 'author', value: thread1.author },
        ],
        links: [
          { rel: 'deleteAction', href: thread1.deleteUrl },
          { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
          {
            rel: 'comment',
            href: thread1.commentsUrl,
          },
        ],
      },
    ],
  },
};

export const threadWhereCannotCreateCommentFetch = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [
      {
        href: thread1.url,
        data: [
          { name: 'id', value: 'thread1' },
          { name: 'slug', value: thread1.slug },
          { name: 'dateCreated', value: thread1.dateCreated },
          { name: 'dateDeleted', value: '' },
          { name: 'headline', value: thread1.headline },
          { name: 'articleBody', value: thread1.articleBody },
          { name: 'author', value: thread1.author },
        ],
        links: [
          { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
          {
            rel: 'comment',
            href: thread1.commentsUrl,
          },
        ],
      },
    ],
  },
};

export const deletedThreadFetch = {
  url: thread1.url,
  method: 'GET',
  status: 200,
  body: {
    items: [
      {
        href: thread1.url,
        data: [
          { name: 'id', value: 'thread1' },
          { name: 'slug', value: thread1.slug },
          { name: 'dateCreated', value: thread1.dateCreated },
          { name: 'dateDeleted', value: thread1.dateDeleted },
          { name: 'headline', value: thread1.headline },
          { name: 'articleBody', value: '' },
          { name: 'author', value: thread1.author },
        ],
        links: [
          { rel: 'isPartOf', href: 'http://localhost:4000/api/board' },
          {
            rel: 'comment',
            href: thread1.commentsUrl,
          },
        ],
      },
    ],
  },
};

export const createThreadFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'POST',
  status: 201,
  body: {
    items: [api[0]],
  },
};

export const canCreateThreadFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS,POST',
  },
};

export const cannotCreateThreadFetch = {
  url: 'http://localhost:4000/api/threads',
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS',
  },
};

export const deleteThreadFetch = {
  url: thread1.deleteUrl,
  method: 'POST',
  status: 200,
  body: {
    items: deleteThreadApi,
  },
};

export const db = [
  {
    id: 'thread1',
    slug: thread1.slug,
    dateCreated: thread1.dateCreated,
    dateDeleted: null,
    headline: thread1.headline,
    articleBody: thread1.articleBody,
    userId: 'user1',
    boardId: 'board1',
  },
  {
    id: 'thread2',
    slug: thread2.slug,
    dateCreated: thread2.dateCreated,
    dateDeleted: thread2.dateDeleted,
    headline: thread2.headline,
    articleBody: thread2.articleBody,
    userId: 'user1',
    boardId: 'board1',
  },
  {
    id: 'thread3',
    slug: thread3.slug,
    dateCreated: thread3.dateCreated,
    dateDeleted: null,
    headline: thread3.headline,
    articleBody: thread3.articleBody,
    userId: 'user1',
    boardId: 'board1',
  },
];
