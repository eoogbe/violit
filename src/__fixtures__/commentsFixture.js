/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as THREADS from './threadsFixture';
import * as USER from './userFixture';

export const comment1 = {
  id: 'comment1',
  dateCreated: '2019-10-09T10:11:12.123Z',
  dateDeleted: '2019-10-09T12:11:12.123Z',
  text: 'Sed do eiusmod tempor incididunt ut labore et dolore.',
  threadSlug: THREADS.thread1.slug,
  author: USER.username,
};

export const comment2 = {
  id: 'comment2',
  dateCreated: '2019-10-08T07:08:09.101Z',
  dateDeleted: '2019-10-08T07:09:09.101Z',
  text: 'Ut enim ad minim veniam.',
  threadSlug: THREADS.thread1.slug,
  author: USER.username,
};

export const comment3 = {
  id: 'comment3',
  dateCreated: '2019-09-30T01:02:03.456Z',
  dateDeleted: '2019-10-03T01:02:03.456Z',
  text: 'Quis nostrud exercitation ullamco laboris nisi ut.',
  threadSlug: THREADS.thread1.slug,
  author: USER.username,
};

export const createInput = {
  id: 'comment1',
  text: comment1.text,
  threadSlug: THREADS.thread1.slug,
};

export const normalized = {
  comment1: {
    id: 'comment1',
    dateCreated: comment1.dateCreated,
    dateDeleted: null,
    text: comment1.text,
    threadSlug: comment1.threadSlug,
    author: comment1.author,
  },
  comment2: {
    id: 'comment2',
    dateCreated: comment2.dateCreated,
    dateDeleted: comment2.dateDeleted,
    text: comment2.text,
    threadSlug: comment2.threadSlug,
    author: comment2.author,
  },
  comment3: {
    id: 'comment3',
    dateCreated: comment3.dateCreated,
    dateDeleted: null,
    text: comment3.text,
    threadSlug: comment3.threadSlug,
    author: comment3.author,
  },
};

export const unnormalized = [
  normalized.comment1,
  normalized.comment2,
  normalized.comment3,
];

export const api = [
  {
    href: 'http://localhost:4000/api/comments/comment1',
    data: [
      { name: 'id', value: 'comment1' },
      { name: 'dateCreated', value: comment1.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'text', value: comment1.text },
      { name: 'author', value: comment1.author },
      { name: 'threadSlug', value: THREADS.thread1.slug },
    ],
    links: [
      {
        rel: 'deleteAction',
        href: 'http://localhost:4000/api/comments/comment1/delete',
      },
      { rel: 'about', href: THREADS.thread1.url },
    ],
  },
  {
    href: 'http://localhost:4000/api/comments/comment2',
    data: [
      { name: 'id', value: 'comment2' },
      { name: 'dateCreated', value: comment2.dateCreated },
      { name: 'dateDeleted', value: comment2.dateDeleted },
      { name: 'text', value: comment2.text },
      { name: 'author', value: comment2.author },
      { name: 'threadSlug', value: THREADS.thread1.slug },
    ],
    links: [
      { rel: 'about', href: THREADS.thread1.url },
    ],
  },
  {
    href: 'http://localhost:4000/api/comments/comment3',
    data: [
      { name: 'id', value: 'comment3' },
      { name: 'dateCreated', value: comment3.dateCreated },
      { name: 'dateDeleted', value: '' },
      { name: 'text', value: comment3.text },
      { name: 'author', value: comment3.author },
      { name: 'threadSlug', value: THREADS.thread1.slug },
    ],
    links: [
      {
        rel: 'deleteAction',
        href: 'http://localhost:4000/api/comments/comment3/delete',
      },
      { rel: 'about', href: THREADS.thread1.url },
    ],
  },
];

export const deleteCommentApi = [
  {
    href: 'http://localhost:4000/api/comments/comment1/delete',
    data: [
      { name: 'actionStatus', value: 'completedActionStatus' },
      { name: 'endTime', value: comment1.dateDeleted },
    ],
    links: [
      {
        rel: 'object',
        href: 'http://localhost:4000/api/comments/comment1',
      },
      { rel: 'targetCollection', href: THREADS.thread1.commentsUrl },
    ],
  },
];

export const commentsFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'GET',
  status: 200,
  body: {
    items: api,
  },
};

export const createCommentFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'POST',
  status: 201,
  body: {
    items: [api[0]],
  },
};

export const canCreateCommentFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS,POST',
  },
};

export const cannotCreateCommentFetch = {
  url: THREADS.thread1.commentsUrl,
  method: 'OPTIONS',
  status: 204,
  headers: {
    Allow: 'GET,HEAD,OPTIONS',
  },
};

export const deleteCommentFetch = {
  url: 'http://localhost:4000/api/comments/comment1/delete',
  method: 'POST',
  status: 200,
  body: {
    items: deleteCommentApi,
  },
};

export const db = [
  {
    id: 'comment1',
    dateCreated: comment1.dateCreated,
    dateDeleted: null,
    content: comment1.text,
    userId: 'user1',
    threadId: 'thread1',
  },
  {
    id: 'comment2',
    dateCreated: comment2.dateCreated,
    dateDeleted: comment2.dateDeleted,
    content: comment2.text,
    userId: 'user1',
    threadId: 'thread1',
  },
  {
    id: 'comment3',
    dateCreated: comment3.dateCreated,
    dateDeleted: null,
    content: comment3.text,
    userId: 'user1',
    threadId: 'thread1',
  },
];
