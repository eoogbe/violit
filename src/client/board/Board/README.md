```jsx
import { MemoryRouter } from 'react-router-dom';
import * as BOARD from '__fixtures__/boardFixture';
import * as THREADS from '__fixtures__/threadsFixture';

mockFetch([
  BOARD.boardFetch,
  THREADS.threadsFetch,
]);

<MemoryRouter>
  <Board />
</MemoryRouter>
```
