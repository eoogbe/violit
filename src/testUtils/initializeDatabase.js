/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import path from 'path';
import Database from 'better-sqlite3';
import bcrypt from 'bcrypt';
import * as BOARD from '__fixtures__/boardFixture';
import * as COMMENTS from '__fixtures__/commentsFixture';
import * as THREADS from '__fixtures__/threadsFixture';
import * as USER from '__fixtures__/userFixture';

const INSERT_USER_SQL =
`INSERT INTO user (id, username, password)
VALUES (:id, :username, :password)`;

const INSERT_BOARD_SQL =
`INSERT INTO board (id, name)
VALUES (:id, :name)`;

const INSERT_THREAD_SQL =
`INSERT INTO thread (
  id,
  slug,
  dateCreated,
  dateDeleted,
  headline,
  articleBody,
  userId,
  boardId
) VALUES (
  :id,
  :slug,
  :dateCreated,
  :dateDeleted,
  :headline,
  :articleBody,
  :userId,
  :boardId
)`;

const INSERT_COMMENT_SQL =
`INSERT INTO comment (id, dateCreated, dateDeleted, content, userId, threadId)
VALUES (:id, :dateCreated, :dateDeleted, :content, :userId, :threadId)`;

/** Initializes the test database. */
export default () => {
  const dbPath = path.resolve(__dirname, '..', '..', 'data', 'test.db');
  const db = new Database(dbPath, {
    fileMustExist: true,
  });

  db.prepare('DELETE FROM comment').run();
  db.prepare('DELETE FROM thread').run();
  db.prepare('DELETE FROM board').run();
  db.prepare('DELETE FROM user').run();

  db.prepare(INSERT_USER_SQL).run({
    id: 'user1',
    username: USER.username,
    password: bcrypt.hashSync(USER.password, 4),
  });

  db.prepare(INSERT_BOARD_SQL).run(BOARD.board);

  const insertThreadStmt = db.prepare(INSERT_THREAD_SQL);
  THREADS.db.forEach((thread) => {
    insertThreadStmt.run(thread);
  });

  const insertCommentStmt = db.prepare(INSERT_COMMENT_SQL);
  COMMENTS.db.forEach((comment) => {
    insertCommentStmt.run(comment);
  });

  return db;
};
