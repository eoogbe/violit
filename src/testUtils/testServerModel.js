/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as R from 'ramda';
import * as BOARD from '__fixtures__/boardFixture';
import * as COMMENTS from '__fixtures__/commentsFixture';
import * as THREADS from '__fixtures__/threadsFixture';
import * as USER from '__fixtures__/userFixture';

const CAN_DELETE_THREAD_RESULTS = {
  [THREADS.thread1.slug]: true,
  [THREADS.thread2.slug]: false,
  [THREADS.thread3.slug]: true,
};

const CAN_DELETE_COMMENT_RESULTS = {
  comment1: true,
  comment2: false,
  comment3: true,
};

/** Creates a test server model with default values. */
export const createModel = R.mergeDeepRight({
  auth: {
    sign: () => Promise.resolve('token123'),
    getCredentials: () => Promise.resolve(null),
    createCredentials: () => Promise.resolve('token123'),
  },
  board: {
    getBoard: () => BOARD.board,
  },
  threads: {
    getThreads: () => THREADS.unnormalized,
    getThreadById: (id) => ({
      id,
      slug: THREADS.thread1.slug,
      dateCreated: THREADS.thread1.dateCreated,
      headline: THREADS.thread1.headline,
      articleBody: THREADS.thread1.articleBody,
      author: THREADS.thread1.author,
    }),
    getThreadBySlug: (slug) => ({
      slug,
      id: 'thread1',
      dateCreated: THREADS.thread1.dateCreated,
      headline: THREADS.thread1.headline,
      articleBody: THREADS.thread1.articleBody,
      author: THREADS.thread1.author,
    }),
    createThread: () => THREADS.normalized.thread1,
    softDeleteThread: () => ({
      id: 'thread1',
      slug: THREADS.thread1.slug,
      dateCreated: THREADS.thread1.dateCreated,
      dateDeleted: THREADS.thread1.dateDeleted,
      headline: THREADS.thread1.headline,
      articleBody: THREADS.thread1.articleBody,
      author: THREADS.thread1.author,
    }),
    canDeleteThread: (thread) => CAN_DELETE_THREAD_RESULTS[thread.slug],
  },
  comments: {
    getCommentsByThreadSlug: () => COMMENTS.unnormalized,
    getCommentById: (id) => ({
      ...COMMENTS.normalized.comment1,
      id,
    }),
    createComment: () => COMMENTS.normalized.comment1,
    softDeleteComment: (id) => ({
      id,
      dateCreated: COMMENTS.comment1.dateCreated,
      dateDeleted: COMMENTS.comment1.dateDeleted,
      text: COMMENTS.comment1.text,
      threadSlug: COMMENTS.comment1.threadSlug,
      author: COMMENTS.comment1.author,
    }),
    canDeleteComment: (comment) => CAN_DELETE_COMMENT_RESULTS[comment.id],
  },
  users: {
    getUserById: () => USER.credentials,
    getUserByUsername: (username) => ({
      ...USER.credentials,
      username,
    }),
    createUser: () => Promise.resolve(USER.credentials),
  },
});

/** Creates a test server model for an authenticated user. */
export const createAuthenticatedModel =
  R.mergeDeepRight(createModel({
    auth: {
      getCredentials: () => Promise.resolve(USER.credentials),
    },
  }));
