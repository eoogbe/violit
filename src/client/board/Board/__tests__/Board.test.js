/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import * as BOARD from '__fixtures__/boardFixture';
import * as THREADS from '__fixtures__/threadsFixture';
import mockFetch from 'testUtils/mockFetch';
import { wait } from 'testUtils/wait';
import Board from '../Board';

const NOW = 1572566400000;

const expectRender = async (responses) => {
  Date.now = jest.fn().mockReturnValue(NOW);
  mockFetch(responses);

  let renderer;
  await TestRenderer.act(async () => {
    renderer = TestRenderer.create(
      <MemoryRouter>
        <Board />
      </MemoryRouter>
    );

    await wait();
  });

  return expect(renderer.toJSON()).toMatchSnapshot();
};

it('renders without crashing', async () => {
  return expectRender([
    BOARD.boardFetch,
    THREADS.threadsFetch,
  ]);
});

it('sets the document title', async () => {
  mockFetch([
    BOARD.boardFetch,
    THREADS.threadsFetch,
  ]);

  await TestRenderer.act(async () => {
    TestRenderer.create(
      <MemoryRouter>
        <Board />
      </MemoryRouter>
    );
  });

  expect(document.title).toBe('Violit');
});
