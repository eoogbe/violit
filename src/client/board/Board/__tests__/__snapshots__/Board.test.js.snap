// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<article
  className="sc-fBuWsC bJYtKI"
>
  <header
    className="sc-fMiknA kcrmtG"
  >
    <h2
      className="sc-dVhcbM hquhcZ"
    >
      dance
    </h2>
    <a
      href="/new-thread"
      onClick={[Function]}
    >
      <svg
        color="#7F00FF"
        fill="currentColor"
        height="1.5em"
        size="1.5em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#7F00FF",
          }
        }
        title="New thread"
        viewBox="0 0 24 24"
        width="1.5em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </a>
  </header>
  <ul
    className="sc-jhAzac gBCyfg"
  >
    <li
      className="sc-Rmtcm hQsHdC"
    >
      <a
        className="sc-bRBYWo hGkBnq"
        href="/threads/lorem-ipsum"
        onClick={[Function]}
      >
        <article
          itemScope={true}
          itemType="http://schema.org/DiscussionForumPosting"
        >
          <header
            className="sc-jKJlTe dCQNcG"
          >
            <span
              className="sc-eNQAEJ jsbjCq"
              itemProp="author"
            >
              fred-astaire
            </span>
            <time
              dateTime="2019-09-29T01:23:45.679Z"
              itemProp="dateCreated"
              title="2019-09-29T01:23:45.679Z"
            >
              1 month ago
            </time>
          </header>
          <p
            className="sc-hzDkRC knCefW"
            itemProp="headline"
          >
            Lorem ipsum
          </p>
        </article>
      </a>
    </li>
    <li
      className="sc-Rmtcm hQsHdC"
    >
      <a
        className="sc-bRBYWo hGkBnq"
        href="/threads/commodo-consequat"
        onClick={[Function]}
      >
        <article
          itemScope={true}
          itemType="http://schema.org/DiscussionForumPosting"
        >
          <header
            className="sc-jKJlTe dCQNcG"
          >
            <span
              className="sc-eNQAEJ jsbjCq"
              itemProp="author"
            >
              fred-astaire
            </span>
            <time
              dateTime="2019-09-27T03:21:09.876Z"
              itemProp="dateCreated"
              title="2019-09-27T03:21:09.876Z"
            >
              1 month ago
            </time>
          </header>
          <p
            className="sc-hzDkRC knCefW"
            itemProp="headline"
          >
            Commodo consequat
          </p>
        </article>
      </a>
    </li>
  </ul>
</article>
`;
